src/bp-templates/bp-nouveau/includes/activity/template-tags.php

```diff
@@ -272,21 +273,127 @@ function bp_nouveau_activity_state() {
 			</span>
 		</a>
 		<span class="ac-state-separator">&middot;</span>
-		<a href="#" class="activity-state-comments">
-			<span class="comments-count">
-				<?php
-				if ( $comment_count > 1 ) {
-					echo $comment_count . ' ' . __( 'Comments', 'buddyboss' );
-				} else {
-					echo $comment_count . ' ' . __( 'Comment', 'buddyboss' );
-				}
-				?>
-			</span>
-		</a>
+		<?php if ( bp_activity_can_comment() ) : ?>
+			<a href="#" class="activity-state-comments">
+				<span class="comments-count">
+					<?php
+					if ( $comment_count > 1 ) {
+						echo $comment_count . ' ' . __( 'Comments', 'buddyboss' );
+					} else {
+						echo $comment_count . ' ' . __( 'Comment', 'buddyboss' );
+					}
+					?>
+				</span>
+			</a>
+		<?php endif; ?>
+
+		<?php if ( ! bp_activity_can_comment() ) : ?>
+			<div href="#" class="activity-state-comments">
+				<span class="comments-count">
+					<?php
+					if ( $comment_count > 1 ) {
+						echo $comment_count . ' ' . __( 'Comments', 'buddyboss' );
+					} else {
+						echo $comment_count . ' ' . __( 'Comment', 'buddyboss' );
+					}
+					?>
+				</span>
+			</div>
+		<?php endif; ?>
+		
 	</div>
 	<?php
 }
 
+/**
+ * Output the action buttons inside an Activity inner content.
+ *
+ * @since BuddyBoss 1.7.1
+ *
+ * @param array $args See bp_nouveau_wrapper() for the description of parameters.
+ */
+function bp_nouveau_activity_inner_buttons( $args = array() ) {
```

Comment:
```suggestion
function bb_nouveau_activity_inner_buttons( $args = array() ) {
```

Reply:
@chetansatasiya 

Change it here https://github.com/buddyboss/buddyboss-platform/pull/2290/commits/fc6d7f59c599309d3572fd697e0db53442958579